### PR TITLE
Allow multiple marblerun instances to be installed with helm

### PR DESCRIPTION
**Pull request by OverOrion**
### Proposed changes
Based on #368 this PR allows multiple Marblerun deployments in the same Kubernetes cluster, but with different namespaces.

~Currently the user needs to create the `namespace` and create the secret manually:~
EDIT: it works without manually creating it.

### How the `charts/README.md` should be edited?
We used the following helm command to install marblerun:
```bash
helm install marblerun ./charts/ --create-namespace -n $namespace \
        --set marbleInjector.useCertManager=true \
        --set marbleInjector.start=true \
        --set coordinator.hostname=$host
        --set dcap.qpl=intel \
        --set dcap.pccsUrl=https://$host:8081/sgx/certification/v3/ \
        --set dcap.useSecureCert=FALSE \
        -f ./charts/values.yaml 

```
Should it be added to the `README.md`?

### CODE REVIEW

File: charts/templates/coordinator.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
     app.kubernetes.io/component: coordinator
     app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
     app.kubernetes.io/name: coordinator
-    app.kubernetes.io/part-of: marblerun
+    app.kubernetes.io/part-of: {{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:10]:
```suggestion
    app.kubernetes.io/part-of: marblerun
```
There already is a label to filter by namespace (`{{ .Values.global.coordinatorNamespaceLabel }}: {{ .Release.Namespace }}`)
This label is supposed to allow application level filter, its not scoped to a namespace.

File: charts/templates/coordinator.yaml [hunk 2]
```diff
@@ -131,10 +131,10 @@ metadata:
     app.kubernetes.io/component: client-api
     app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
     app.kubernetes.io/name: coordinator-client-api
-    app.kubernetes.io/part-of: marblerun
+    app.kubernetes.io/part-of: {{ .Release.Namespace }}
     app.kubernetes.io/version: {{ .Values.coordinator.version }}
 spec:
-  type: ClusterIP
+  type: NodePort
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:137]: Why this change?
There should be no namespace issues with ClusterIP

File: charts/templates/marble-injector.yaml [hunk 3]
```diff
@@ -75,7 +75,7 @@ spec:
 apiVersion: admissionregistration.k8s.io/v1
 kind: MutatingWebhookConfiguration
 metadata:
-  name: marble-injector
+  name: marble-injector-{{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/marble-injector.yaml:78]: Actually had this part in my local changes, missed pushing it
Thanks for catching :+1:

**Review by daniel-weisse — CHANGES_REQUESTED** [review]

**Comment by daniel-weisse** [comment]
Changed base to my PR, should make review a bit easier

> Currently the user needs to create the namespace and create the secret manually

Is this regarding the current state on main, or the state of my PR?
Because my PR should already fix that issue by using cert-manager to provision it

**Comment by OverOrion** [comment]
> Changed base to my PR, should make review a bit easier
> 
> > Currently the user needs to create the namespace and create the secret manually
> 
> Is this regarding the current state on main, or the state of my PR? Because my PR should already fix that issue by using cert-manager to provision it

We have written an install script before your PR and we were working on it at the same time, I just removed the manual secret creation step from the script and it works as well! :)

File: charts/templates/coordinator.yaml (hunk 1 above)
Comment (OverOrion) [reply charts/templates/coordinator.yaml:10]: You are right, nice catch!

File: charts/templates/coordinator.yaml (hunk 2 above)
Comment (OverOrion) [reply charts/templates/coordinator.yaml:137]: Sorry for that, it is a leftover.  While we were testing it it was easier for us to have that port externally available.

File: charts/templates/coordinator.yaml [hunk 4]
```diff
@@ -23,7 +23,7 @@ spec:
         app.kubernetes.io/component: coordinator
         app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
         app.kubernetes.io/name: coordinator
-        app.kubernetes.io/part-of: marblerun
+        app.kubernetes.io/part-of: {{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:26]:
```suggestion
        app.kubernetes.io/part-of: marblerun
```
These should also just be labeled `marblerun`

File: charts/templates/coordinator.yaml [hunk 5]
```diff
@@ -110,7 +110,7 @@ metadata:
     app.kubernetes.io/component: persistent-storage
     app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
     app.kubernetes.io/name: coordinator-pv-claim
-    app.kubernetes.io/part-of: marblerun
+    app.kubernetes.io/part-of: {{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:113]:
```suggestion
    app.kubernetes.io/part-of: marblerun
```

File: charts/templates/coordinator.yaml [hunk 6]
```diff
@@ -131,7 +131,7 @@ metadata:
     app.kubernetes.io/component: client-api
     app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
     app.kubernetes.io/name: coordinator-client-api
-    app.kubernetes.io/part-of: marblerun
+    app.kubernetes.io/part-of: {{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:134]:
```suggestion
    app.kubernetes.io/part-of: marblerun
```

File: charts/templates/coordinator.yaml [hunk 7]
```diff
@@ -152,7 +152,7 @@ metadata:
     app.kubernetes.io/component: mesh-api
     app.kubernetes.io/created-by: {{ .Values.global.createdBy }}
     app.kubernetes.io/name: coordinator-mesh-api
-    app.kubernetes.io/part-of: marblerun
+    app.kubernetes.io/part-of: {{ .Release.Namespace }}
```
Comment (daniel-weisse) [inline charts/templates/coordinator.yaml:155]:
```suggestion
    app.kubernetes.io/part-of: marblerun
```

File: charts/templates/coordinator.yaml (hunk 4 above)
Comment (OverOrion) [reply charts/templates/coordinator.yaml]: Thanks again, fixed in the latest fixup commit.
